include/ginkgo/core/matrix/batch_ell.hpp

```diff
@@ -289,6 +290,34 @@ class BatchEll
         return this->get_const_col_idxs()[this->linearize_index(0, row, idx)];
     }
 
+    /**
+     * Creates an immutable BatchEll matrix from a set of constant arrays.
+     *
+     * @param exec  the executor to create the matrix on
+     * @param size  the dimensions of the matrix
+     * @param values  the value array of the matrix
+     * @param col_idxs  the column index array of the matrix
+     * @param num_stored_elems_per_row  the number of stored nonzeros per row
+     * @param stride  the column-stride of the value and column index array
+     * @returns A smart pointer to the constant matrix wrapping the input arrays
+     *          (if they reside on the same executor as the matrix) or a copy of
+     *          the arrays on the correct executor.
+     */
+    static std::unique_ptr<const BatchEll> create_const(
+        std::shared_ptr<const Executor> exec, const batch_dim<2>& size,
+        const batch_stride& num_stored_elems_per_row,
+        const batch_stride& stride,
+        gko::detail::ConstArrayView<ValueType>&& values,
+        gko::detail::ConstArrayView<IndexType>&& col_idxs)
+    {
+        // cast const-ness away, but return a const object afterwards,
+        // so we can ensure that no modifications take place.
+        return std::unique_ptr<const BatchEll>(new BatchEll{
+            exec, size, num_stored_elems_per_row.at(0), stride.at(0),
```

Comment:
Do you need to get the value at 0 ? I think passing in the batch_dim and batch_stride should also work ?